internal/abci/abci.go

```diff
@@ -738,13 +746,15 @@ func (a *AbciApp) FinalizeBlock(ctx context.Context, req *abciTypes.RequestFinal
 	}
 
 	networkParams.MigrationStatus = a.chainContext.NetworkParameters.MigrationStatus
-
 	// store any changes to the network params
 	err = meta.StoreDiff(ctx, a.consensusTx, &oldNetworkParams, networkParams)
 	if err != nil {
 		return nil, fmt.Errorf("failed to store network params diff: %w", err)
 	}
 
+	// update the last migration status
+	a.lastMigrationStatus = networkParams.MigrationStatus
+
```

Comment:
Once again, you are just duplicating an update that we already make to the other lastMigrationStatus variable on line 748. Seems totally redundant.

Reply:
Not actually, with migration status, the migration status is updated generally in the NotifyHeight, which occurs after we update the consensus params. 

Finalize Block
- old network params
- process txs
- update network params
- precommit 
- NotifyHeight(changesets) ---> this is where the network params get updated, so u are never factoring this in the diff.

Reply:
Why dont we just move `NotifyHeight` before recommit? We don't need/use the changesets in `NotifyHeight`